chatsky/core/vector_store.py

```diff
@@ -0,0 +1,188 @@
+from typing import Optional, List, Dict, Any
+from pydantic import BaseModel
+import logging
+from uuid import uuid4
+from context import Context
+from message import Message
+from service import Service
+from langchain.vectorstores import Chroma
+from langchain.embeddings import HuggingFaceEmbeddings
+from langchain.schema import Document
+
+logger = logging.getLogger(__name__)
+
+
+class VectorStoreConfig(BaseModel):
+    """Configuration model for vector store initialization."""
+
+    collection_name: str = "default"
+    """Name of the collection in vector database. Defaults to "default"."""
+    top_k: int = 3
+    """Number of top results to return during retrieval. Defaults to 3."""
+    persist_directory: Optional[str] = None
+    """Directory path for persistent storage. Defaults to None."""
+    model_name: str = "DeepPavlov/rubert-base-cased"
+    """Name of the embedding model. Defaults to "DeepPavlov/rubert-base-cased"."""
+
+
+class VectorStoreService:
+    """
+    Abstract base class for vector store services.
+
+    Provides common interface for document storage and retrieval operations.
+    Must be subclassed with specific vector database implementations.
+    """
+
+    def __init__(self, config: VectorStoreConfig):
+        """
+        Initialize the vector store service with configuration.
+
+        :param config: Configuration parameters for the vector store.
+        """
+        self.config = config
+        self.initialized = False
+
+    async def initialize(self, ctx: Context):
+        """
+        Initialize connection to the vector database and register with context.
+
+        :param ctx: Framework context object where the vector store will be registered.
+
+        :raises RuntimeError: If initialization fails.
+        """
+        if not self.initialized:
+            ctx.framework_data.vector_store = self
+            self.initialized = True
+            logger.info("Vector store initialized")
+
+    async def add_documents(self, documents: List[Message], metadata: dict[Optional]):
```

Comment:
It's better to have consistency in typing (at least in your own code):
Either both should be typed as `List` and `Dict` or `list` and `dict`.

Chatsky uses `List` and `Dict` from typing.